weatherman.py

```diff
@@ -0,0 +1,80 @@
+"""
+Generating reports of temperature from given text files.
+"""
+
+import os
+import argparse
+import sys
+
+file_list = os.listdir('./weatherdata/')
+hottest_day = {}
+hottest_date = {}
+coldest_day = {}
+maxi_humidity= {}
+mini_humidity= {}
+
+parser = argparse.ArgumentParser(prog='weatherman'
+								, usage='%(prog)s [report#] [data_dir]'
+								, description='Generate weather reports')
+parser.add_argument('-d','--data_dir', type = str, metavar = ''
+					, help = 'Directory containing weather data files'
+					, required=True)
+parser.add_argument('-r','--report_number', type = str, metavar = ''
+					, choices = ['1', '2'], help = '1 for Annual Max/Min Temperature, 2 for Hottest day of each year'
+					, required=True)
+try:
+    args = parser.parse_args()	
+except:
+    parser.print_help()
+    sys.exit(0)
+
+
+def print_report(report_number):
+
+	if report_number == '1':
+		print("Year		MAX Temp	MIN Temp	MAX Humidity	MIN Humidity")
+		for key in sorted(hottest_day):	
+			print(key, "		", hottest_day[key], " 		", coldest_day[key], "		", maxi_humidity[key], "		", mini_humidity[key])
+
+	elif report_number == '2':
+		print("Year		MAX Temp	DATE")
+		for key in sorted(hottest_day):	
+			print(key, "		", hottest_day[key], "		", hottest_date[key])
+
+def file_path(data_dir):
+	for file_name in file_list:
+
+		with open(data_dir + file_name) as data_file:
+			lines = data_file.read().splitlines()[2:-1]
+			for line in lines:
+				line = line.split(',')
+				year = int(line[0].split('-')[0])
+				date = line[0]
+				try:
+					max_temp = int(line[1])
+					min_temp = int(line[3])
+					max_humid= int(line[7])
+					min_humid = int(line[9])
```

Comment:
if exception occur due to any of above castings data in the whole row is not being processed.

Reply:
left.

Reply:
Fixed.